src/Struct/FindologicProduct.php

```diff
@@ -1036,55 +967,27 @@ protected function parseCategoryAttributes(
                 continue;
             }
 
-            $seoUrls = $this->fetchCategorySeoUrls($categoryEntity);
-            if ($seoUrls->count() > 0) {
-                foreach ($seoUrls->getElements() as $seoUrlEntity) {
-                    $catUrl = $seoUrlEntity->getSeoPathInfo();
-                    if (!Utils::isEmpty($catUrl)) {
-                        $catUrls[] = $this->getCatUrlPrefix() . sprintf('/%s', ltrim($catUrl, '/'));
-                    }
-                }
-            }
-
-            $catUrl = sprintf(
-                '/%s',
-                ltrim(
-                    $this->router->generate(
-                        'frontend.navigation.page',
-                        ['navigationId' => $categoryEntity->getId()],
-                        RouterInterface::ABSOLUTE_PATH
-                    ),
-                    '/'
-                )
-            );
-
-            if (!Utils::isEmpty($catUrl)) {
-                $catUrls[] = $catUrl;
-            }
-
             $categoryPath = Utils::buildCategoryPath(
                 $categoryEntity->getBreadcrumb(),
                 $this->navigationCategory
             );
 
             if (!Utils::isEmpty($categoryPath)) {
-                $categories[] = $categoryPath;
+                $categories = array_merge($categories, [$categoryPath]);
+            }
+
+            // We only export "cat_urls", and "cat" recursively if there is Direct Integration, otherwise we only
+            // export "cat" without recursion.
+            if ($this->isDirectIntegration()) {
+                $catUrls = array_merge($catUrls, $this->urlBuilderService->buildCatUrls($categoryEntity));
+                $parentCategory = explode('_', $categoryPath);
+                $categories = array_merge($categories, $parentCategory);
             }
         }
     }
 
-    protected function getCatUrlPrefix(): string
+    private function isDirectIntegration(): bool
```

Comment:
Never add `private` methods in here, otherwise plugin extensions have no way of overriding or calling them.

Reply:
Please check if there is any way to tell our linter to disallow private methods in a specific class.

Reply:
No there is not. There is one that checks if visibility is there or not but I didnt find anything to define specific visibility for a specific class